wagtail_wordpress_import/block_builder_defaults.py

```diff
@@ -0,0 +1,290 @@
+    If the image can be retrived from the remote site and saved into a Wagtail ImageModel
+    the soup is modified.
+    """
+    soup = BeautifulSoup(html, "html.parser")
+    images = soup.find_all("img")
+    for image in images:
+        if image.attrs and image.attrs["src"]:
+            image_src = get_abolute_src(image.attrs["src"], conf_domain_prefix())
+            saved_image = get_or_save_image(image_src)
+            if saved_image:
+                image_embed = soup.new_tag("embed")
+                image_embed.attrs["embedtype"] = "image"
+                image_embed.attrs["id"] = saved_image.id
+                image_embed.attrs["alt"] = get_image_alt(image)
+                image_embed.attrs["format"] = get_alignment_class(image)
+                image.replace_with(image_embed)
+        else:
+            print(f"IMAGE HAS NO SRC: {image}")
+
+    return str(soup)
+
+
+def get_image_alt(img_tag):
+    return img_tag.attrs["alt"] if "alt" in img_tag.attrs else None
+
+
+def get_image_file_name(src):
+    return src.split("/")[-1] if src else None  # need the last part
+
+
+def image_exists(name):
+    try:
+        return ImportedImage.objects.get(title=name)
+    except ImportedImage.DoesNotExist:
+        pass
+
+
+def conf_get_requests_settings():
+    return getattr(
+        settings,
+        "WAGTAIL_WORDPRESS_IMPORTER_REQUESTS_SETTINGS",
+        {
+            "headers": {"User-Agent": "WagtailWordpressImporter"},
+            "timeout": 1,
+            "stream": False,
+        },
+    )
+
+
+def get_or_save_image(src):
+    image_file_name = get_image_file_name(src)
+    existing_image = image_exists(image_file_name)
+    if not existing_image:
+        response, valid, type = fetch_url(src)
+        if valid and (type in conf_valid_image_content_types()):
+            temp_image = NamedTemporaryFile(delete=True)
+            temp_image.name = image_file_name
+            temp_image.write(response.content)
+            temp_image.flush()
+            retrieved_image = ImportedImage(
+                file=File(file=temp_image), title=image_file_name
+            )
+            retrieved_image.save()
+            temp_image.close()
+            return retrieved_image
+        else:
+            print(f"RECEIVED INVALID RESPONSE: {src}")
+    return existing_image
+
+
+def fetch_url(src, r=None, status=False, content_type=None):
+    """general purpose url fetcher with ability to pass in own config"""
+    try:
+        r = requests.get(src, **conf_get_requests_settings())
+        status = r.status_code == 200
+        content_type = (
+            r.headers["content-type"].lower() if r.headers.get("content-type") else ""
+        )
+    except requests.ConnectTimeout:
+        print(f"THERE WAS A PROBLEM WITH REQUESTS FETCHING: {src}")
+    return r, status, content_type
+
+
+def get_abolute_src(src, domain_prefix=None):
+    src = re.sub("^\/+", "", src)
```

Comment:
**Suggestion**: Use `.lstrip()` instead of regex substitution. (I think I've seen and suggested the same elsewhere for readability.)

```suggestion
    src = src.lstrip("/")
```